addons/website_event_track/models/event_track_stage.py

```diff
@@ -37,3 +34,14 @@ class TrackStage(models.Model):
     legend_normal = fields.Char(
         'Grey Kanban Label', default=lambda s: _('In Progress'), translate=True, required=True,
         help='Override the default value displayed for the normal state for kanban selection.')
+
+    @api.onchange('is_cancel')
+    def _onchange_cancel(self):
```

Comment:
Better make a computed editable field. Onchange are not a reliable way of implementing that kind of logic.